uxarray/grid.py

```diff
@@ -99,6 +99,9 @@ def __init__(self, dataset, **kwargs):
         # initialize convenience attributes
         self.__init_grid_var_attrs__()
 
+        # build face dimension
+        self._build_face_dimension()
```

Comment:
An expensive process for large grids.
If the target problem solved does not need face_dimension this will cause unnecessary overhead. 

We can think about guarding it later. Please put a TODO comment about it.